src/ext/oblivc/ot.c

```diff
@@ -717,6 +742,28 @@ recverExtensionBoxNew (ProtocolDesc* pd, int srcParty, int keyBytes)
   }
   return r;
 }
+
+RecverExtensionBox*
+recverExtensionBoxSplit (ProtocolDesc * pd, RecverExtensionBox* t)
+{
+  const int k = t->keyBytes*8;
+  RecverExtensionBox* r = malloc(sizeof *r);
+  r->srcParty = t->srcParty;
+  r->pd = pd;
+  r->keyBytes=t->keyBytes;
+  char seed[OT_SEEDLEN];
+  int i;
+  r->keyblock0 = malloc(sizeof(BCipherRandomGen*[k]));
+  r->keyblock1 = malloc(sizeof(BCipherRandomGen*[k]));
+  for(i=0;i<k;++i)
+  { randomizeBuffer(t->keyblock0[i], seed, OT_SEEDLEN);
+    r->keyblock0[i] = newBCipherRandomGenByKey(seed);
+    randomizeBuffer(t->keyblock1[i], seed, OT_SEEDLEN);
+    r->keyblock1[i] = newBCipherRandomGenByKey(seed);
+  }
```

Comment:
Oh ... wait this is actually a more significant change than I thought. Yeah, I'm not so sure this particular method is secure. However, it may be possible to use a variation of this that works. Instead of having the old keyblocks continue with their old seeds, we may want to reseed them like this:

```C
randomizeBuffer(t->keyblock0[i], seed1, OT_SEEDLEN);
randomizeBuffer(t->keyblock0[i], seed2, OT_SEEDLEN);
r->keyblock0[i] = newBCipherRandomGenByKey(seed1);
resetBCipherRandomGen(t->keyblock0[i], seed2);
```

The same for `keyblock1`, and the sender side. It requires a full security proof, which I have not yet done, but I'm fairly confident this tweak will fix it. I'll try to fill out the proof when I can, but feel free to have a swing if you have time. It may be a good excuse for you to learn proving MPC protocols if you haven't picked it up yet ;)

This may or may not make the whole nonce thing redundant, I haven't decided.